libraries/AP_GPS/AP_GPS.h

```diff
@@ -481,11 +481,9 @@ class AP_GPS
         uint8_t current_baud;
         bool auto_detected_baud;
         struct UBLOX_detect_state ublox_detect_state;
-#if !HAL_MINIMIZE_FEATURES
```

Comment:
This does increase the build size for minimized boards.

Reply:
no, I don't think it does (as its a union)

Reply:
I checked, it does change the build size here, and this is not a union, it's a struct. (Each driver type needs it's own detect state))

Reply:
With the `HAL_MINIMIZE_FEATURES` check:
```
BUILD SUMMARY
Build directory: /home/wickedshell/code/ardupilot/build/fmuv2
Target         Text    Data  BSS     Total  
--------------------------------------------
bin/arduplane  952124  3960  193284  1149368
```
Without the `HAL_MINIMIZE_FEATURES` check:
```
BUILD SUMMARY
Build directory: /home/wickedshell/code/ardupilot/build/fmuv2
Target         Text    Data  BSS     Total  
--------------------------------------------
bin/arduplane  952172  3960  193284  1149416

```

Reply:
sorry about that, of course its a struct
now, the Q is if 50 bytes is worth it ...

Reply:
The other option is to just move the detect state into AP_GPS.cpp as a static struct (or static member of `AP_GPS::detect_instance()`) which would allow the maintenance of the `HAL_MINIMIZE_FEATURES` check, without polluting the header.

Reply:
Make builds already don't fit for PX4-FMUv2, so we're probably approaching it too with waf builds.